pex/resolver.py

```diff
@@ -657,6 +668,55 @@ def _spawn_install(
         )
         return SpawnedJob.wait(job=install_job, result=install_result)
 
+    def _resolve_direct_file_deps(
+        self,
+        install_requests,  # type: Iterable[InstallRequest]
+        max_parallel_jobs=None,  # type: Optional[int]
+        analyzed=None,  # type: Optional[OrderedDict[ProjectName, InstallRequest]]
+    ):
+        # type: (...) -> Iterable[InstallRequest]
+
+        already_analyzed = (
+            analyzed or OrderedDict()
+        )  # type: OrderedDict[ProjectName, InstallRequest]
+        build_requests = OrderedSet()  # type: OrderedSet[BuildRequest]
+        for install_request in install_requests:
+            metadata = DistMetadata.load(install_request.wheel_path)
+            for requirement in metadata.requires_dists:
+                if requirement.project_name in already_analyzed:
+                    continue
+                if not requirement.url:
+                    continue
+                urlinfo = urlparse.urlparse(requirement.url)
+                if urlinfo.scheme != "file":
+                    continue
+                dist_path = unquote(urlinfo.path).rstrip()
+                if not os.path.exists(dist_path):
+                    raise Unsatisfiable(
+                        "The {wheel} wheel has a dependency on {url} which does not exist on this "
+                        "machine.".format(wheel=install_request.wheel_file, url=requirement.url)
+                    )
+                if dist_path.endswith(".whl"):
+                    already_analyzed[requirement.project_name] = InstallRequest.create(
```

Comment:
Hrm, that's not right. This dep is not already analyzed - it still needs to hae its dist metadata loaded and its requirements processed. Fix coming...

Reply:
Ok, the issue was Pip actually. Now with recursion tested.